### PR TITLE
Deprecate old Process class

### PR DESCRIPTION
**Pull request by ngkavin**
This PR adds `process_deprecated.py`. This file contains the `Process` class that provides backwards compatibility with echopype version 0.4.1. The new `Process` class saves the new `EchoData` object and passes it to the new Calibrate and Preprocess functions for calibration, calculating MVBS, and removing noise.

### CODE REVIEW

File: echopype/echodata/echodata.py
Comment (lsetiawan) [inline echopype/echodata/echodata.py:200]: What does this do? Isn't `converted_raw_path` either ends with `.nc` or `.zarr`?

Comment (ngkavin) [reply echopype/echodata/echodata.py:200]: When converting to nc then to  zarr, we lose the .nc filepath so nc_path would have a .zarr extension. Also, having access to both names was supported in the old version.

Comment (lsetiawan) [reply echopype/echodata/echodata.py:200]: Great! Thanks for those explanation :)

**Comment by leewujung** [comment]
@ngkavin : thanks for the PR!
Except for the `.astype(np.int64)` problem I think we can merge this once the other smaller things are changed.
The revamping tests work #303 will be the tests for the linkages you put in here under `process_deprecated.py` .

**Comment by emiliom** [comment]
I'm pasting my comment from Slack, vis a vis the data type for frequency:

> In the convention document, I now see that there are four frequency variables (with units of Hz) listed. All of them are defined to be **float** type. I would've guessed an int type, too. Geez, I guess frequencies defined as tenths of a Hz are not unusual? Anyway, that makes the int64 vs int32 issue less ... um, relevant?

**Comment by leewujung** [comment]
> I'm pasting my comment from Slack, vis a vis the data type for frequency:
> 
> > In the convention document, I now see that there are four frequency variables (with units of Hz) listed. All of them are defined to be **float** type. I would've guessed an int type, too. Geez, I guess frequencies defined as tenths of a Hz are not unusual? Anyway, that makes the int64 vs int32 issue less ... um, relevant?

Ugh, my bad, I just assumed this to be one of those variables that are supposed to be `int32`. Frequency should be `float` to allow fractions. 

@ngkavin could you look into where in the AZFP convertor that encoded `frequency` to be `int`? We will need that to be in a different PR to keep a record of that issue for the release note.

File: echopype/preprocess/noise_est.py
Comment (leewujung) [reply echopype/preprocess/noise_est.py:34]: Alright, let's resolve #307 via a PR based on @emiliom 's comments below and then revert this change here before merging this PR. :)

Comment (leewujung) [reply echopype/preprocess/noise_est.py]: #309 resolves this.

**Review by leewujung — APPROVED** [review]

**Pull request merged**
